package/network/utils/umbim/files/lib/netifd/proto/mbim.sh

```diff
@@ -84,93 +89,143 @@ _proto_mbim_setup() {
 	echo "mbim[$$]" "Checking pin"
 	umbim $DBG -n -t $tid -d $device pinstate || {
 		echo "mbim[$$]" "PIN required"
-		proto_notify_error "$interface" PIN_FAILED
-		proto_block_restart "$interface"
-		return 1
+		if [ ! "$ignore_pinfail" ] || [ "$ignore_pinfail" = 0 ]; then
```

Comment:
I don't think you should make this a config option.  It's just unnecessary hassle when configuring these modems.  And worse: You do not want to ignore PIN1 failures.

Parse the "required pin:" value and DTRT depening on what it is.

Reply:
i guess i'll take a look inside umbim cli and see what possible responses there are :/
what is pin2? is it specific to sierra? is it possible pin2 could be a sim pin on a secondary sim on some device that supports it that isn't sierra?
it's hard to know what exactly we should be looking for here... should i also ignore and proceed on "Not supported"?

Reply:
> what is pin2?

It's a PIN code locking other SIM features than network access.  I tried to Google for the correct definition, but this proved somewhat difficult. The best I found was this: https://www.nttdocomo.co.jp/english/support/trouble/password/pin/

PIN2 is a standard SIM feature. It is not Sierra Wireless specific.  But requesting it when you don't actually use any SIM features requiring it seems to be SWI specific.

I suggest looking specifically for PIN1 unlocking requests and ignoring anything else.  You probably don't want a script like this to handle PUK1 or PUK2 codes :-)